projects/packages/lazy-images/tests/php/test_class.lazy-images.php

```diff
@@ -604,19 +604,58 @@ public function add_skip_lazy_class_to_attributes( $attributes ) {
 
 	/**
 	 * Confirms that the lazy images module is not loaded with Gutenberg 16.6.
+	 *
+	 * @covers Jetpack_Lazy_Images::should_force_deactivate
+	 * @dataProvider get_should_force_deactivate_data
+	 *
+	 * @param array $version_details Version details (WP and Gutenberg).
+	 * @param bool  $expected        Whether or not the module should be deactivated.
 	 */
-	public function test_lazy_images_not_loaded_with_gutenberg_16_6() {
-		// Verify without the constant set that the instance is returned.
-		$this->assertInstanceOf( Jetpack_Lazy_Images::class, Jetpack_Lazy_Images::instance() );
-
-		// Verify loading on old versions of Gutenberg.
-		Constants::set_constant( 'IS_GUTENBERG_PLUGIN', true );
-		Constants::set_constant( 'GUTENBERG_VERSION', '16.5.0' );
-		$this->assertInstanceOf( Jetpack_Lazy_Images::class, Jetpack_Lazy_Images::instance() );
-
-		// Set the Gutenberg constants and test instance.
-		Constants::set_constant( 'GUTENBERG_VERSION', '16.6.0' );
-		$this->assertNull( Jetpack_Lazy_Images::instance() );
+	public function test_should_force_deactivate( $version_details, $expected ) {
+		global $wp_version;
+		$previous_version = $wp_version;
+
+		Constants::set_constant( 'IS_GUTENBERG_PLUGIN', $version_details['gutenberg'] );
+		Constants::set_constant( 'GUTENBERG_VERSION', $version_details['gutenberg_version'] );
+		$wp_version = $version_details['wp']; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		$this->assertSame( $expected, Jetpack_Lazy_Images::should_force_deactivate() );
+
 		Constants::clear_constants();
+		$wp_version = $previous_version; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
```

Comment:
FYI, if the test fails these lines won't be reached to reset things. If you want to fix that, you could use a try-finally or move the resetting logic to a `tear_down()` method.

Reply:
That's a good call. 9f62b74d690d452c7eee9f1cfba8f920c3395e99